diff --git a/browser/actors/ScreenshotsComponentChild.sys.mjs b/browser/actors/ScreenshotsComponentChild.sys.mjs
index bee089ad4d..88680ca112 100644
--- a/browser/actors/ScreenshotsComponentChild.sys.mjs
+++ b/browser/actors/ScreenshotsComponentChild.sys.mjs
@@ -141,6 +141,9 @@ export class ScreenshotsComponentChild extends JSWindowActorChild {
       case "Screenshots:Download":
         this.requestDownloadScreenshot(event.detail.region);
         break;
+      case "Screenshots:ZenAddToBoard":
+        this.requestZenAddToBoard(event.detail);
+        break;
       case "Screenshots:OverlaySelection": {
         let { hasSelection, overlayState } = event.detail;
         this.sendOverlaySelection({ hasSelection, overlayState });
@@ -195,8 +198,20 @@ export class ScreenshotsComponentChild extends JSWindowActorChild {
       region,
     });
     this.endScreenshotsOverlay({ doNotResetMethods: true });
   }
 
+  /**
+   * Send a request to upload the screenshots to zen board
+   *
+   * @param {object} detail The detail object from the overlay event
+   */
+  requestZenAddToBoard(detail) {
+    const { region, anchor } = detail;
+    region.devicePixelRatio = this.contentWindow.devicePixelRatio;
+    this.sendAsyncMessage("Screenshots:ZenAddToBoard", { region, anchor });
+    this.endScreenshotsOverlay({ doNotResetMethods: true });
+  }
+
   getDocumentTitle() {
     return this.document.title;
   }
