diff --git a/browser/components/screenshots/ScreenshotsOverlayChild.sys.mjs b/browser/components/screenshots/ScreenshotsOverlayChild.sys.mjs
index 7f35341fe05175336bd339b39be1853e4a73b773..30e206d3f7f2c114016ef6c7957c62b046cb7c0e 100644
--- a/browser/components/screenshots/ScreenshotsOverlayChild.sys.mjs
+++ b/browser/components/screenshots/ScreenshotsOverlayChild.sys.mjs
@@ -91,6 +91,7 @@ export class ScreenshotsOverlay {
       downloadAttributes,
       copyAttributes,
       previewFaceAriaLabel,
+      screenshotsZenAddToBoardAttributes,
     ] = lazy.overlayLocalization.formatMessagesSync([
       { id: "screenshots-cancel-button" },
       { id: "screenshots-component-cancel-button" },
@@ -104,6 +105,7 @@ export class ScreenshotsOverlay {
         args: { shortcut: copyShorcut },
       },
       { id: "screenshots-overlay-preview-face-label" },
+      { id: "screenshots-zen-add-to-board" },
     ]);
 
     return `
@@ -171,6 +173,7 @@ export class ScreenshotsOverlay {
               <button id="cancel" class="screenshots-button" title="${cancelAttributes.attributes[0].value}" aria-label="${cancelAttributes.attributes[1].value}"><img/></button>
               <button id="copy" class="screenshots-button" title="${copyAttributes.attributes[0].value}" aria-label="${copyAttributes.attributes[1].value}"><img/><label>${copyAttributes.value}</label></button>
               <button id="download" class="screenshots-button primary" title="${downloadAttributes.attributes[0].value}" aria-label="${downloadAttributes.attributes[1].value}"><img/><label>${downloadAttributes.value}</label></button>
+              <button id="zen-add-to-board" class="screenshots-button" title="${screenshotsZenAddToBoardAttributes.value}" aria-label="${screenshotsZenAddToBoardAttributes.value}"><img/><label>${screenshotsZenAddToBoardAttributes.value}</label></button>
             </div>
           </div>
         </div>
@@ -416,9 +419,29 @@ export class ScreenshotsOverlay {
       case "download":
         this.downloadSelectedRegion();
         break;
+      case "zen-add-to-board":
+        this.zenAddToBoard(event);
+        break;
     }
   }

+  zenAddToBoard(event) {
+    this.#dispatchEvent("Screenshots:ZenAddToBoard", {
+      anchor: { x: event.screenX, y: event.screenY },
+      region: {
+        left: this.selectionRegion.dimensions.left,
+        top: this.selectionRegion.dimensions.top,
+        right: this.selectionRegion.dimensions.right,
+        bottom: this.selectionRegion.dimensions.bottom,
+        width: this.selectionRegion.dimensions.width,
+        height: this.selectionRegion.dimensions.height,
+        devicePixelRatio: this.window.devicePixelRatio,
+        viewportWidth: this.windowDimensions.dimensions.clientWidth,
+        viewportHeight: this.windowDimensions.dimensions.clientHeight,
+      },
+    });
+  }
+
   maybeCancelScreenshots() {
     if (this.#state === STATES.CROSSHAIRS) {
       this.#dispatchEvent("Screenshots:Close", {
